atlassian_jwt_auth/contrib/django_app/auth.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+import atlassian_jwt_auth
+import jwt
+import requests
+
+from django.conf import settings
+
+from .models import Issuer
+
+
+logger = logging.getLogger(__name__)
+
+
+class ASAPAuthBackend():
+    def authenticate(token):
+        verifier = _get_verifier()
+        allowed_aud = settings.ASAP_ALLOWED_AUDIENCE
+
+        try:
+            claims = verifier.verify_jwt(
+                token, allowed_aud, leeway=60
+            )
+        except (
+            requests.exceptions.HTTPError,
+            requests.exceptions.ConnectionError,
+            jwt.Exceptions.InvalidTokenError
+        ):
+            logging.exception('Authentication failed while verifying ASAP token')
+            return None
+
+        try:
+            return Issuer.objects.get(issuer=claims['iss']).user
```

Comment:
`Issuer.user` represents a specific Django user.

In Django, all authentication backends are required to take some sort of credential and return a matching user. The best way to do this in the context of ASAP is to validate the token, extract the issuer, and then look up the `iss` value and determine if it matches a user. If Django finds a user linked to that `iss` (via `Issuer`), the user is authenticated. If Django does not find a user, authentication fails even though the token is valid (since the service doesn't recognize the issuer)